cmd/controller-manager/app/controllermanager.go

```diff
@@ -170,6 +171,11 @@ func init() {
 func startClusterController(ctx controllerscontext.Context) (enabled bool, err error) {
 	mgr := ctx.Mgr
 	opts := ctx.Opts
+
+	if err := cluster.IndexField(mgr); err != nil {
+		return false, err
+	}
```

Comment:
The indexer is only used by the taint manager, right? 
So it should be set only when `--enable-taint-manager=true`.

Reply:
Agree, now only taint manager need this index field. If needed by other controllers in the future, we could remove this check.